tiledb/sm/query/query.cc

```diff
@@ -836,6 +899,48 @@ Status Query::set_buffer(
   return reader_.set_buffer(name, buffer, buffer_size, check_null_buffers);
 }
 
+Status Query::set_data_buffer(
+    const std::string& name,
+    void* const buffer,
+    uint64_t* const buffer_size,
+    const bool check_null_buffers) {
+  RETURN_NOT_OK(check_set_fixed_buffer(name));
+
+  if (type_ == QueryType::WRITE)
+    return writer_.set_buffer_data(name, buffer, buffer_size);
+  return reader_.set_buffer_data(name, buffer, buffer_size, check_null_buffers);
+}
+
+Status Query::set_off_buffer(
+    const std::string& name,
+    uint64_t* const buffer_off,
+    uint64_t* const buffer_off_size,
+    const bool check_null_buffers) {
+  RETURN_NOT_OK(check_set_fixed_buffer(name));
+
+  if (type_ == QueryType::WRITE)
+    return writer_.set_buffer_offsets(name, buffer_off, buffer_off_size);
+  return reader_.set_buffer_offsets(
+      name, buffer_off, buffer_off_size, check_null_buffers);
+}
+
+Status Query::set_val_buffer(
```

Comment:
same here, we want to say "validity" instead of "val" so that it is not confused with "value"